cobalt-parser/src/parser/expr/literal.rs

```diff
@@ -0,0 +1,506 @@
+use std::{borrow::Cow, collections::HashMap, str::Chars};
+
+use cobalt_ast::{
+    ast::{ErrorAST, IntLiteralAST, StringLiteralAST, StructLiteralAST},
+    BoxedAST,
+};
+use cobalt_errors::{CobaltError, ParserFound, SourceSpan};
+
+use crate::{
+    lexer::tokens::{Delimiter, LiteralToken, TokenKind},
+    parser::Parser,
+    utils::CharBytesIterator,
+};
+
+impl<'src> Parser<'src> {
+    /// Parses a literal expression, e.g. `1`, `"hello"`, `true`, etc.
+    ///
+    /// Going into this function, the current token should be the first token
+    /// of this grammar.
+    ///
+    /// ```text
+    /// literal := LITERAL
+    /// ```
+    pub(crate) fn parse_literal(&mut self) -> (BoxedAST<'src>, Vec<CobaltError<'src>>) {
+        assert!(self.current_token.is_some());
+
+        let span = self.current_token.unwrap().span;
+        let mut suffix = None;
+        let mut errors = vec![];
+
+        match self.current_token.unwrap().kind {
+            TokenKind::Literal(LiteralToken::Int(s)) => {
+                self.next();
+
+                // Check if the next token is a type hint.
+
+                if self.current_token.is_some() {
```

Comment:
Parsed suffixes are for the AST to handle. In the future, there may be custom suffixes added, and this will be necessary.

Reply:
Should I change this then, or will it still work?

Reply:
You should change it to allow any valid identifier. That should be as simple as just removing the check.

Reply:
Fixed by [02b3b6c](https://github.com/cobalt-language/cobalt-lang/pull/115/commits/02b3b6cce6c5cca5c5c189e297dcbef9fdf1d273)